File: src/main/java/no/nav/nav/sbl/dialogarena/modiabrukerdialog/api/service/SakerService.java
package no.nav.nav.sbl.dialogarena.modiabrukerdialog.api.service;

import no.nav.nav.sbl.dialogarena.modiabrukerdialog.api.domain.Sak;
import no.nav.nav.sbl.dialogarena.modiabrukerdialog.api.domain.Saker;

import java.util.List;

public interface SakerService {
    public Saker hentSaker(String fnr);

    public List<Sak> hentListeAvSaker(String fnr);

    void knyttBehandlingskjedeTilSak(String fnr, String behandlingskjede, Sak sak) throws Exception;
}

File: api/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/api/service/saker/SakerService.kt
package no.nav.sbl.dialogarena.modiabrukerdialog.api.service.saker

import no.nav.sbl.dialogarena.modiabrukerdialog.api.domain.saker.Sak
import no.nav.sbl.dialogarena.modiabrukerdialog.api.exceptions.JournalforingFeilet

interface SakerService {
    class Resultat(val saker: MutableList<Sak> = mutableListOf(), val feiledeSystemer: MutableList<String> = mutableListOf())

    fun hentSaker(fnr: String): Resultat

    @Deprecated("")
    fun hentSammensatteSaker(fnr: String): List<Sak>

    @Deprecated("")
    fun hentPensjonSaker(fnr: String): List<Sak>

    @Throws(JournalforingFeilet::class)
    fun knyttBehandlingskjedeTilSak(fnr: String?, behandlingskjede: String?, sak: Sak, enhet: String?)
}
